### ISSUE TITLE
Exercise #2 Magic number Checkstyle + "static" modifer not allowed 

### ISSUE THREAD

**Issue by airrehn** [nus-cs2030/2223-s1#97]
Hi! May I know how you implement the variables like "... every 50 passengers" or "... per 40 metres.." in _Exercise 2 level 5_ ? Without running into the magic number error on Checkstyle? 

I found out if I declared them in the child class (_BigCruise_) itself like `private final static int a = 40`, then use the `a`, this compiles and run as usual and removes the magic number error on codestyle but gives me a new error on Checkstyle,

> 'static' modifier out of order with the JLS suggestions.

If I remove the static modifer, then I get a compilation error saying 

> error: cannot reference `a` before supertype constructor has been called. 

Really confused how to go about this styling error. My code runs as expected but just have troubles submitting on codecrunch because of these errors. Any help would be appreciated, thank you!

**Comment by MingFengC** [comment]
Hi I believe your first error can be fixed by declaring it as `private static final int A = 40`.

You may refer to this page for their specifications: https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html

**Comment by hongming321** [comment]
I think if you are declaring a constant with `private static final` the variable must be named in all Upper Case alphabets.

**Comment by airrehn** [comment]
> Hi I believe your first error can be fixed by declaring it as `private static final int A = 40`.
> 
> You may refer to this page for their specifications: https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html

Hi! That worked! Thank you so much. Yeah `static` must come before `final`. Hope the others will see this!

**Comment by airrehn** [comment]
> I think if you are declaring a constant with `private static final` the variable should be named in all Upper Case alphabets.

Yeah haha sorry, I overlooked the correct casing in this post, it was just an example, my bad haha.

**Comment by sansuthn** [comment]
@MingFengC  This worked! Thanks alot!
